update to rxjs@6.0.0-alpha.4

IgorMinar opened the pull request:
Bam!

This PR is much bigger than typical RxJS code because of a unusual way we used RxJS in the past in order to reduce payload size and not introduce Observable.prototype patching due to Angular's use of RxJS.

IgorMinar committed:
feat(core): upgrade rxjs to 6.0.0-alpha.4

mary-poppins commented:
You can preview 6802b13 at https://pr22573-6802b13.ngbuilds.io/.

mary-poppins commented:
You can preview fd9fce3 at https://pr22573-fd9fce3.ngbuilds.io/.

mary-poppins commented:
You can preview f65c06d at https://pr22573-f65c06d.ngbuilds.io/.

mary-poppins commented:
You can preview 2d2cb68 at https://pr22573-2d2cb68.ngbuilds.io/.

mary-poppins commented:
You can preview 8aa2240 at https://pr22573-8aa2240.ngbuilds.io/.

mary-poppins commented:
You can preview 7eaef06 at https://pr22573-7eaef06.ngbuilds.io/.

mary-poppins commented:
You can preview 9ca6cde at https://pr22573-9ca6cde.ngbuilds.io/.

mary-poppins commented:
You can preview 0a6810e at https://pr22573-0a6810e.ngbuilds.io/.

mary-poppins commented:
You can preview b04941f at https://pr22573-b04941f.ngbuilds.io/.

mary-poppins commented:
You can preview 14b229a at https://pr22573-14b229a.ngbuilds.io/.

mary-poppins commented:
You can preview de2aca5 at https://pr22573-de2aca5.ngbuilds.io/.

mary-poppins commented:
You can preview eac4697 at https://pr22573-eac4697.ngbuilds.io/.

mary-poppins commented:
You can preview 0dc689b at https://pr22573-0dc689b.ngbuilds.io/.

mary-poppins commented:
You can preview 95bca52 at https://pr22573-95bca52.ngbuilds.io/.

mary-poppins commented:
You can preview 660d839 at https://pr22573-660d839.ngbuilds.io/.

mary-poppins commented:
You can preview 27948f5 at https://pr22573-27948f5.ngbuilds.io/.

mary-poppins commented:
You can preview 4d865d2 at https://pr22573-4d865d2.ngbuilds.io/.

mary-poppins commented:
You can preview 57c4b81 at https://pr22573-57c4b81.ngbuilds.io/.

mary-poppins commented:
You can preview 12e5112 at https://pr22573-12e5112.ngbuilds.io/.

mary-poppins commented:
You can preview d60123e at https://pr22573-d60123e.ngbuilds.io/.

alfaproject commented on packages/bazel/src/ng_package/ng_package.bzl:
```diff
@@ -18,8 +18,8 @@ WELL_KNOWN_GLOBALS = {
-    "rxjs": "Rx",
-    "rxjs/operators": "Rx.Observable.prototype",
+    'rxjs': 'Rx',
+    'rxjs/operators': 'Rx.operators',
```
Inconsistent quotes

mary-poppins commented:
You can preview c0eeac3 at https://pr22573-c0eeac3.ngbuilds.io/.

gkalpak committed:
build(aio): temporarily use RxJS from root `node_modules/` when using local packages

mary-poppins commented:
You can preview c0bf0ad at https://pr22573-c0bf0ad.ngbuilds.io/.

mary-poppins commented:
You can preview 9ca7815 at https://pr22573-9ca7815.ngbuilds.io/.

mary-poppins commented:
You can preview fd06926 at https://pr22573-fd06926.ngbuilds.io/.

mary-poppins commented:
You can preview 35f7975 at https://pr22573-35f7975.ngbuilds.io/.

mary-poppins commented:
You can preview 4ff78e2 at https://pr22573-4ff78e2.ngbuilds.io/.

mary-poppins commented:
You can preview f39302c at https://pr22573-f39302c.ngbuilds.io/.

mary-poppins commented:
You can preview eaad5b8 at https://pr22573-eaad5b8.ngbuilds.io/.

mary-poppins commented:
You can preview 0145924 at https://pr22573-0145924.ngbuilds.io/.

mary-poppins commented:
You can preview b3be4e4 at https://pr22573-b3be4e4.ngbuilds.io/.

IgorMinar commented on aio/content/examples/http/src/app/config/config.service.ts:
```diff
@@ -82,8 +81,8 @@ export class ConfigService {
-    // return an ErrorObservable with a user-facing error message
-    return new ErrorObservable(
+    // return an observable with a user-facing error message
+    return throwError(
```
For some reason this doesn't seem to work. The symptom I see is that the observable that subscribes to the result of `getConfig` (https://github.com/angular/angular/pull/22573/files#diff-447af6fbd9dbfe9744627f202c8440c0R33) doesn't get the 'Something bad happened;...' string via the error channel.

benlesh replied:
Where is it being called? My guess is you're handling the error somewhere else, because that function is pretty simple... basically `new Observable(s => s.error(yourError))`

IgorMinar replied:
resolved via a fix of a bad import in rxjs

mary-poppins commented:
You can preview 9bc1884 at https://pr22573-9bc1884.ngbuilds.io/.

mary-poppins commented:
You can preview 90521dc at https://pr22573-90521dc.ngbuilds.io/.

mary-poppins commented:
You can preview 229c64d at https://pr22573-229c64d.ngbuilds.io/.

mary-poppins commented:
You can preview 55682c6 at https://pr22573-55682c6.ngbuilds.io/.

mary-poppins commented:
You can preview e271538 at https://pr22573-e271538.ngbuilds.io/.

mary-poppins commented:
You can preview 49e5103 at https://pr22573-49e5103.ngbuilds.io/.

mary-poppins commented:
You can preview 72361ea at https://pr22573-72361ea.ngbuilds.io/.

IgorMinar committed:
ci: temporarily increase payload size limit for cli-hello-world until we update cli to v6

mary-poppins commented:
You can preview 500d85c at https://pr22573-500d85c.ngbuilds.io/.

mary-poppins commented:
You can preview b81f782 at https://pr22573-b81f782.ngbuilds.io/.

mary-poppins commented:
You can preview 1c175c5 at https://pr22573-1c175c5.ngbuilds.io/.

mary-poppins commented:
You can preview 358995d at https://pr22573-358995d.ngbuilds.io/.

mary-poppins commented:
You can preview b3125ef at https://pr22573-b3125ef.ngbuilds.io/.

mary-poppins commented:
You can preview 67b6d36 at https://pr22573-67b6d36.ngbuilds.io/.

mary-poppins commented:
You can preview 5e3983d at https://pr22573-5e3983d.ngbuilds.io/.

mary-poppins commented:
You can preview 32b4be9 at https://pr22573-32b4be9.ngbuilds.io/.

IgorMinar commented on packages/compiler/test/aot/test_util.ts:
```diff
@@ -59,6 +59,8 @@ export interface EmitterOptions {
 function calcPathsOnDisc() {
   const moduleFilename = module.filename.replace(/\\/g, '/');
+  // TODO(i): this is suspicious because it relies on build.sh output
+  //   which is problematic when we are running tests under bazel - review with Chuck
```
@chuckjaz do you know why is the path below hardcoded in this test and only conditionally used? it seems a bit odd.

mary-poppins commented:
You can preview d425889 at https://pr22573-d425889.ngbuilds.io/.

mary-poppins commented:
You can preview 527e225 at https://pr22573-527e225.ngbuilds.io/.

mary-poppins commented:
You can preview c8ee781 at https://pr22573-c8ee781.ngbuilds.io/.

mary-poppins commented:
You can preview a51b6ae at https://pr22573-a51b6ae.ngbuilds.io/.

mary-poppins commented:
You can preview 73a53a7 at https://pr22573-73a53a7.ngbuilds.io/.

mary-poppins commented:
You can preview 42f4625 at https://pr22573-42f4625.ngbuilds.io/.

mary-poppins commented:
You can preview eaf9c5f at https://pr22573-eaf9c5f.ngbuilds.io/.

mary-poppins commented:
You can preview 6fabd38 at https://pr22573-6fabd38.ngbuilds.io/.

mary-poppins commented:
You can preview 0ad5d43 at https://pr22573-0ad5d43.ngbuilds.io/.

mary-poppins commented:
You can preview 60c6477 at https://pr22573-60c6477.ngbuilds.io/.

mary-poppins commented:
You can preview be17c9a at https://pr22573-be17c9a.ngbuilds.io/.

mary-poppins commented:
You can preview 7d4a6a0 at https://pr22573-7d4a6a0.ngbuilds.io/.

mary-poppins commented:
You can preview 5497093 at https://pr22573-5497093.ngbuilds.io/.

mary-poppins commented:
You can preview 290dbe0 at https://pr22573-290dbe0.ngbuilds.io/.

mary-poppins commented:
You can preview 63f0f08 at https://pr22573-63f0f08.ngbuilds.io/.

mary-poppins commented:
You can preview 3352e95 at https://pr22573-3352e95.ngbuilds.io/.

mary-poppins commented:
You can preview 8203e03 at https://pr22573-8203e03.ngbuilds.io/.

mary-poppins commented:
You can preview 64b1653 at https://pr22573-64b1653.ngbuilds.io/.

mary-poppins commented:
You can preview 6eb6ec9 at https://pr22573-6eb6ec9.ngbuilds.io/.

mary-poppins commented:
You can preview 2285d1c at https://pr22573-2285d1c.ngbuilds.io/.

IgorMinar committed:
ci: temporarily disable offline_compiler_test.sh

mary-poppins commented:
You can preview 06555e6 at https://pr22573-06555e6.ngbuilds.io/.

mary-poppins commented:
You can preview f22c86d at https://pr22573-f22c86d.ngbuilds.io/.

mary-poppins commented:
You can preview 1aea83c at https://pr22573-1aea83c.ngbuilds.io/.

mary-poppins commented:
You can preview e0bc680 at https://pr22573-e0bc680.ngbuilds.io/.

IgorMinar committed:
ci: temporarily disable tests that depend on angular-in-memory-web-api

angular-in-memory-web-api is not yet compatible with rxjs v6 and rxjs v6 backwards compatibility package is not yet ready to be used.

mary-poppins commented:
You can preview ad72391 at https://pr22573-ad72391.ngbuilds.io/.

mary-poppins commented:
You can preview f3f787c at https://pr22573-f3f787c.ngbuilds.io/.

mary-poppins commented:
You can preview b6c467c at https://pr22573-b6c467c.ngbuilds.io/.

mary-poppins commented:
You can preview ec6ff11 at https://pr22573-ec6ff11.ngbuilds.io/.

jasonaden commented on packages/router/src/router_config_loader.ts:
```diff
@@ -6,12 +6,10 @@
-import {map} from 'rxjs/operator/map';
-import {mergeMap} from 'rxjs/operator/mergeMap';
+import {Compiler, InjectionToken, Injector, NgModuleFactory, NgModuleFactoryLoader} from '@angular/core';
+// TODO(i): switch to fromPromise once it's expored in rxjs
```
remove

jasonaden reviewed (approved):
Ignore comments for simplification changes. Will tack on to an add-on PR.

I removed a couple TODOs & pushed a fixup commit. Looks good!

mary-poppins commented:
You can preview 8b0784e at https://pr22573-8b0784e.ngbuilds.io/.

mary-poppins commented:
You can preview 246285b at https://pr22573-246285b.ngbuilds.io/.

benlesh reviewed (changes requested):
Recommend this goes to beta.0 instead.

mary-poppins commented:
You can preview ecfdbaa at https://pr22573-ecfdbaa.ngbuilds.io/.

IgorMinar committed:
build(aio): temporarily pin elements to 6.0.0-beta.8

the next step will be to update all of aio to 6.0.0-rc.0

mary-poppins commented:
You can preview 6427180 at https://pr22573-6427180.ngbuilds.io/.
